e2e/e2e_test.go

```diff
@@ -136,6 +141,116 @@ func Test_No_Ethereum_TX_Submission(t *testing.T) {
 	verifyDatabaseRecords(setupEnv.DbValidation, expectedTxRecord, receivedSignatures, t)
 }
 
+func Test_E2E_Token_Transfer(t *testing.T) {
+	setupEnv := setup.Load()
+
+	memo := fmt.Sprintf("%s-0-0", receiverAddress)
+
+	wTokenReceiverAddress := common.HexToAddress(receiverAddress)
+
+	// Step 1 - Verify the transfer of Hbars to the Bridge Account
+	transactionResponse, wrappedTokenBalanceBefore := verifyTokenTransferToBridgeAccount(setupEnv, memo, wTokenReceiverAddress, t)
+
+	// Step 2 - Verify the submitted topic messages
+	_, receivedSignatures := verifyTopicMessages(setupEnv, transactionResponse, 0, t)
+
+	// tokenAddress, _ := setup.ParseToken(setupEnv.Clients.RouterContract, setupEnv.TokenID.String())
+
+	transactionData, tokenAddress := verifyTransactionData(setupEnv, transactionResponse, receivedSignatures, t)
+
+	txHash := executeTransaction(setupEnv, transactionResponse, transactionData, tokenAddress, t)
+
+	validateTransactionReceipt(setupEnv, txHash, t)
+
+	validateTokenBalance(setupEnv, wrappedTokenBalanceBefore, wTokenReceiverAddress, t)
+}
```

Comment:
We should add one more step to verify DB records after execution. See the e2e test above.

```Go
// Step 7 - Verify Database records
expectedTxRecord := prepareExpectedTransfer(
		setupEnv.Clients.RouterContract,
		transactionResponse.TransactionID,
		nativeToken,
		"0",
		"0",
		database.ExpectedStatuses{
			Status:          entity_transfer.StatusCompleted,
			StatusSignature: entity_transfer.StatusSignatureMined,
		},
		ethTransactionHash,
		false, t)
	verifyDatabaseRecords(setupEnv.DbValidation, expectedTxRecord, receivedSignatures, t)
```